### ISSUE TITLE
Unexpected assignment of `ptr` when using `HeteroDataBatch.to_homogeneous()`

### ISSUE THREAD

**Issue by ekosman** [pyg-team/pytorch_geometric#6299]
### 🐛 Describe the bug

I try to convert a heterogeneous batch into a homogeneous one, but I encounter the following behavior:

```python
from torch_geometric.data import Batch, HeteroData
import torch

data = []
for i in range(2):
    data_ = HeteroData()
    data_["a"].x = torch.randn(3,3)
    data_["b"].x = torch.randn(3,3)
    data.append(data_)

b = Batch.from_data_list(data).to_homogeneous()

print("Batch:", b.batch)
print("ptr: ", b.ptr)
```

Output:
```
Batch: tensor([0, 0, 0, 1, 1, 1, 0, 0, 0, 1, 1, 1])
ptr:  tensor([0, 3, 6, 0, 3, 6])
```

The resulting `ptr` relates to the batch indices of the original hetero data batch, not the new one.

### Environment

* PyG version: 2.1.0
* PyTorch version: 1.11.0
* OS: ubuntu
* Python version: 3.8
* CUDA/cuDNN version:
* How you installed PyTorch and PyG (`conda`, `pip`, source): conda
* Any other relevant information (*e.g.*, version of `torch-scatter`):

**Comment by rusty1s** [comment]
Thanks for reporting. I fixed it as part of #6304.

**Comment by ekosman** [comment]
Thanks for the extremely fast response!
